### PR TITLE
merge recipe to dev

### PR DESCRIPTION
**Pull request by cnzytari**

### CODE REVIEW

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk1.java [hunk 1]
```diff
@@ -4,23 +4,54 @@
...
+    String input4 = editText4.getText().toString();
 
     //next step button
     public void step2(View view){
+        recipe.add(input1);
+        recipe.add(input2);
+        recipe.add(input3);
+        recipe.add(input4);
+        newRecipe.setTempRecipePass(recipe);
+        x.setTempRecipe(newRecipe);
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk1.java:53]: It doesn't look like newRecipe is being initialized, try calling:
     newRecipe = newRecipe(name); 
     where name is a string for the recipe name

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk1.java [hunk 2]
```diff
@@ -4,23 +4,54 @@
...
+        editText4 = (EditText) findViewById(R.id.editText4);
+
+
         //has no toolbar
 
     }
+    String input1 = editText1.getText().toString();
+    String input2 = editText2.getText().toString();
+    String input3 = editText3.getText().toString();
+    String input4 = editText4.getText().toString();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk1.java:44]: This probably won't do what you want it to do. Try putting lines 41-44 inside the public void step2(View view) function

File: app/src/main/java/letseat/mealdesigner/storage/Recipe.java [hunk 3]
```diff
@@ -46,6 +46,14 @@
      * @return Returns true if successful
      */
     boolean setIngredients(ArrayList<String> ingredients);
+
+    /**
+     * Store the ingredients in the recipe
+     * @param temp An ArrayList of strings where each string passed through the creation walk through
+     * @return Returns true if successful
+     */
+    boolean setTempRecipePass(ArrayList<String> temp);
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/storage/Recipe.java:55]: What does this function do?

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java [hunk 4]
```diff
@@ -6,28 +6,61 @@
...
         startActivity(intent);
     }
     //button to created recipe page
     public void finalCreate(View view){
+        allergies.add(input1);
+        allergies.add(input2);
+        allergies.add(input3);
+        allergies.add(input4);
+        allergies.add(input5);
+        newRecipe.setSteps(allergies);
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java:62]: Currently allergies aren't set up comment line 62 out

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java [hunk 5]
```diff
@@ -6,28 +6,61 @@
...
+        editText3 = (EditText) findViewById(R.id.editText10);
+        editText4 = (EditText) findViewById(R.id.editText11);
+        editText5 = (EditText) findViewById(R.id.editText12);
     }
 
+    final String input1 = editText1.getText().toString();
+    final String input2 = editText2.getText().toString();
+    final String input3 = editText3.getText().toString();
+    final String input4 = editText4.getText().toString();
+    final String input5 = editText5.getText().toString();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java:48]: Put lines 44-48 in finalCreate and remove final from each

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java [hunk 6]
```diff
@@ -6,28 +6,61 @@
...
-
+    private EditText editText1;
+    private EditText editText2;
+    private EditText editText3;
+    private EditText editText4;
+    private EditText editText5;
+    ArrayList<String> allergies  = new ArrayList<>();
+    ArrayAdapter<String> m_adapter;
+    Recipe newRecipe;
+    Database x = (Database) ((MealDesignerApp) getApplication()).getDatabase().getTempRecipe();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java:29]: Change line 29 to Database x = ((MealDesignerApp) getApplication()).getDatabase()

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java [hunk 7]
```diff
@@ -6,28 +6,61 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_recipe_walk6);
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java:34]: Add  newRecipe = x.getTempRecipe() to onCreate

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk5.java [hunk 8]
```diff
@@ -7,10 +7,13 @@
 import android.support.v7.app.AppCompatActivity;
 import android.view.View;
 
+import letseat.mealdesigner.MealDesignerApp;
 import letseat.mealdesigner.R;
 import letseat.mealdesigner.recipies.MainRecipe;
+import letseat.mealdesigner.storage.Database;
 
 public class RecipeWalk5 extends AppCompatActivity {
+    Database x = (Database) ((MealDesignerApp) getApplication()).getDatabase().getTempRecipe();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk5.java:16]: Line 16 not needed

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk4.java [hunk 9]
```diff
@@ -6,24 +6,48 @@
...
 import letseat.mealdesigner.recipies.MainRecipe;
+import letseat.mealdesigner.storage.Database;
+import letseat.mealdesigner.storage.Recipe;
 
 public class RecipeWalk4 extends AppCompatActivity {
-
+    private EditText editText1;
+    ArrayList<String> steps  = new ArrayList<>();
+    Recipe newRecipe;
+    Database x = (Database) ((MealDesignerApp) getApplication()).getDatabase().getTempRecipe();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk4.java:25]: Change line 25 to Database x = ((MealDesignerApp) getApplication()).getDatabase()

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk4.java [hunk 10]
```diff
@@ -6,24 +6,48 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_recipe_walk4);
 
         //has no toolbar
-
+        editText1 = (EditText) findViewById(R.id.editText7);
 
     }
-
+    String step = editText1.getText().toString();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk4.java:35]: put line 35 in addStep

**Review by ghost — APPROVED** [review]
allow to merge it

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java [hunk 11]
```diff
@@ -6,28 +6,61 @@
...
     }
     //button to created recipe page
     public void finalCreate(View view){
+        allergies.add(input1);
+        allergies.add(input2);
+        allergies.add(input3);
+        allergies.add(input4);
+        allergies.add(input5);
+        newRecipe.setSteps(allergies);
+        x.setTempRecipe(newRecipe);
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk6.java:63]: Change line 63 to x.setRecipe(newRecipe);
and add x.clearTemp() below it

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk4.java [hunk 12]
```diff
@@ -6,24 +6,48 @@
...
 public class RecipeWalk4 extends AppCompatActivity {
-
+    private EditText editText1;
+    ArrayList<String> steps  = new ArrayList<>();
+    Recipe newRecipe;
+    Database x = (Database) ((MealDesignerApp) getApplication()).getDatabase().getTempRecipe();
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_recipe_walk4);
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk4.java:30]: add  newRecipe = x.getTempRecipe()  to  onCreate

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk3.java [hunk 13]
```diff
@@ -7,50 +7,80 @@
...
-
             @Override //what to do if no item selected
             public void onNothingSelected(AdapterView<?> parent) {
                 // TODO Auto-generated method stub
             }
         });
     }
+    //get string from fill in boxes
+    String name = editTextName.getText().toString();
+    String amounts = editTextAmount.getText().toString();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk3.java:66]: put lines 65-66 in addIngredient

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk3.java [hunk 14]
```diff
@@ -7,50 +7,80 @@
...
 
 public class RecipeWalk3 extends AppCompatActivity {
-
+    private EditText editTextName;
+    private EditText editTextAmount;
+    ArrayList<String> ingredient  = new ArrayList<>();
+    String unit = "";
+    String finalIngredient;
+    Recipe newRecipe;
+    Database x = (Database) ((MealDesignerApp) getApplication()).getDatabase().getTempRecipe();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk3.java:30]: change line 30 to Database x = ((MealDesignerApp) getApplication()).getDatabase();

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk3.java [hunk 15]
```diff
@@ -7,50 +7,80 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_recipe_walk3);
 
         //has no toolbar
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk3.java:37]: add newRecipe = x.getTempRecipe() to onCreate

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk2.java [hunk 16]
```diff
@@ -54,9 +64,22 @@ public void onNothingSelected(AdapterView<?> parent) {
 
 
     }
-
+    //get string from fill in blank
+    String equipname = editText1.getText().toString();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk2.java:68]: put line 68 in addEquip

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk2.java [hunk 17]
```diff
@@ -7,43 +7,53 @@
...
+import letseat.mealdesigner.storage.Database;
+import letseat.mealdesigner.storage.Recipe;
 
 public class RecipeWalk2 extends AppCompatActivity {
+    private EditText editText1;
+    ArrayList<String> equipment  = new ArrayList<>();
+    Recipe newRecipe;
 
-
+    Database x = (Database) ((MealDesignerApp) getApplication()).getDatabase().getTempRecipe();
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk2.java:29]: change line 29 to Database x = ((MealDesignerApp) getApplication()).getDatabase();

File: app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk2.java [hunk 18]
```diff
@@ -7,43 +7,53 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_recipe_walk2);
```
Comment (shawndur) [inline app/src/main/java/letseat/mealdesigner/recipewalk/RecipeWalk2.java:34]: add newRecipe = x.getTempRecipe to onCreate

**Review by shawndur — COMMENTED** [review]
I left some comments suggesting changes/fixes

**Pull request merged**